packages/api/utils/requestValidation.ts

```diff
@@ -29,3 +32,91 @@ export function shapeValidationForPostRequest<T>(
 
   return new SetupSuccess<T>({ portfolioDraftId }, bodyResult as unknown as T);
 }
+
+interface EnvironmentWithNoAdminProps {
+  appIndex: number;
+  envIndex: number;
+}
+interface AdministratorsFound {
+  hasPortfolioAdminRole: boolean;
+  hasAdminForEachApplication: boolean;
+  hasAdminForEachEnvironment: boolean;
+  applicationsWithNoAdmin: number[];
+  environmentsWithNoAdmin: EnvironmentWithNoAdminProps[];
+  acceptableAdministratorRoles: boolean;
+}
+
+/**
+ * Find and determine if an application step has the correct business rules for
+ * operator admin roles.
+ *
+ * Acceptable admin roles according to the business rules, only one of the following is required:
+ * - 1 root portfolio admin role
+ * - 1 application admin role for each application (without a portfolio admin role above)
+ * - 1 environment admin role for each environment (without a portfolio or application admin role above)
+ *
+ * @param applicationStep - the incoming applicationStep request body
+ * @returns - an object summarizing the admin roles found or missing
+ */
+export function findAdministrators(applicationStep: ApplicationStep): AdministratorsFound {
+  const { operators, applications } = applicationStep;
+  const hasPortfolioAdminRole = adminRoleFound(operators);
+
+  // find apps and envs without admin roles
+  const applicationsWithNoAdmin: number[] = [];
+  const environmentsWithNoAdmin: EnvironmentWithNoAdminProps[] = [];
+  applications.forEach((app, appIndex) => {
+    // apps without admin roles
+    !adminRoleFound(app.operators) && applicationsWithNoAdmin.push(appIndex);
+
+    // env without admin roles
+    app.environments.forEach((env, envIndex) => {
+      !adminRoleFound(env.operators) && environmentsWithNoAdmin.push({ appIndex, envIndex });
+    });
```

Comment:
I am concerned that this is a little too clever, relying on short-circuiting like this. How about:

```
applicationsWithNoAdmin = applications.filter((app) => !adminRoleFound(app.operators));
environmentsWithNoAdmin = applications.flatMap((app) => app.environments).filter((env) => !adminRoleRound(env.operators));
```
?

Reply:
If you don't want to go that route, please switch to using an `if (COND) { }` instead

Reply:
You actually might need something a little more involved like:

```ts
applicationsWithNoAdmin = applications.map((app, appIndex) => { app, appIndex }).filter(({ app }) => !adminRoleFound(app.operators)).map(({app, appIndex}) => appIndex);
```

Reply:
Updated with your more involved suggestion and removed the short-circuiting. 
It was not my intent to be clever with this.